### PR TITLE
Standardize code formatting with clang-format

### LINKED ISSUE #40: Code formatting with clang-format
cuDF uses `clang-format` for code formatting. It would be great if this repo is formatted with the same formatting style as well.

### PR DESCRIPTION
**Pull request by gaohao95**
This PR:
- copy `.clang-format` from cuDF repo
- add a script `run-clang-format.py` to run `clang-format` for all source files
- format all source files using `run-clang-format.py`

close #40

### CODE REVIEW

**Review by nsakharnykh — APPROVED** [review]

File: run-clang-format.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-p", "--path", default="clang-format", help="Path to clang-format executable")
+args = parser.parse_args()
+
+extensions = ["cu", "cpp", "h", "cuh", "hpp"]
+file_paths = []
+
+for extension in extensions:
+    file_paths.extend(glob.glob('**/*.{}'.format(extension), recursive=True))
```
Comment (eyal0) [inline run-clang-format.py:14]: Or in one line:

```
file_paths = [file_path for extension_paths in glob.glob('**/*.{}'.format(extension), recursive=True) for file_path in extension_paths]
```

It's less readable that way but maybe it runs faster.  No need to change this.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard: Cpp11
+StatementMacros:
+  - Q_UNUSED
+  - QT_REQUIRE_VERSION
+# Be consistent with indent-width, even for people who use tab for indentation!
+TabWidth: 2
+UseTab: Never
```
Comment (eyal0) [inline .clang-format:156]: The cudf version has a trailing newline.  Is it necessary or not?

Probably safer to have it.

**Review by eyal0 — APPROVED** [review]
Looks fine with minor, optional, changes.  Thanks!

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,156 @@
+---
+# This file is copied from cuDF repo.
```
Comment (eyal0) [inline .clang-format:2]: Okay, so link to it?

https://github.com/rapidsai/cudf/blob/branch-0.17/cpp/.clang-format

File: run-clang-format.py (hunk 1 above)
Comment (gaohao95) [reply run-clang-format.py:14]: Agree. I like the original way better given that we only have a few extensions and source files, so performance is unlikely to be an issue.

**Pull request merged**